### ISSUE TITLE
Make empty value handling consistent for single field queries

### ISSUE THREAD

**Issue by DrewAPicture** [AffiliateWP/AffiliateWP#2263]
In AffWP core, single field queries have the potential to return unexpected results due to the way `wpdb` treats empty stored values differently in the `get_results()` and `get_col()` methods. This is because of the use of `get_var()` inside of `get_col()`, which defaults to `null` for empty values.

In AffWP, we currently toggle the use of `wpdb->get_results()` with `wpdb->get_col()` for single field queries, such as the following:
```php
// Query
affiliate_wp()->affiliates->get_affiliates( array(
    'fields' => 'rate'
) );
```
In practice, the above code would produce SQL like `SELECT rate FROM affiliates_table`. If we were to run that through `get_results()` right now, we'd see something like this on the result:
```
array(2) {
  [0]=>
  object(stdClass)#1734 (1) {
    ["rate"]=>
    string(0) ""
  }
  [1]=>
  object(stdClass)#1735 (1) {
    ["rate"]=>
    string(0) ""
  }
}
```
However, for single field queries we currently use `get_col()`, which serves to simply retrieve an array of the expected values. Except this is what you would get with that same  query:
```
array(2) {
  [0]=>
  NULL
  [1]=>
  NULL
}
```
One is not like the other :-)

The difference is jarring and goes back to how core has worked from almost the beginning (r112). As we venture into increasingly using field-based queries going forward, being able to write tests with consistency is a must.

In the short term, I've proposed that create a wrapper for `get_results()` that simply plucks the field into an array so that we expect consistent values whether a query is for one or multiple fields. See #2254 for where we implemented multi-field queries. With this new wrapper, running the single field query above would instead result in something like the following:
```
array(2) {
  [0]=>
  string(0) ""
  [1]=>
  string(0) ""
}
```

Note that this would break backward-compatibility for single field queries where the fields were previously empty. We've considered that are willing to accept the break in this context.

Longer term, I've filed a [core trac ticket](https://core.trac.wordpress.org/ticket/40953) to discuss the problem more broadly at the WordPres Core level, 

PR with tests coming up.

**Comment by DrewAPicture** [comment]
Started a PR: #2264 

PR checklist:
* [x] Adjust `get_results()` to use the same `wpdb->get_results()` call for all non-count queries
* [x] Adjust affiliates class to set fields and callback values as expected
* [x] Adjust campaigns class to set fields and callback values as expected
* [x] Adjust creatives class to set fields and callback values as expected
* [x] Adjust payous class to set fields and callback values as expected
* [x] Adjust REST consumers class to set fields and callback values as expected
* [x] Adjust referrals class to set fields and callback values as expected
* [x] Adjust visits class to set fields and callback values as expected
* [ ] Write tests for field, vs fields, vs all (*) to prove callbacks are setting properly
* [x] By virtue of existing tests passing, this tells us the existing fields and general query coverage still works as expected

**Comment by DrewAPicture** [comment]
The upside of this is that we can DRY query handling by using the same `get_results()` call for everything except count queries, then just process the results based on the fields and callback values 👍

**Comment by DrewAPicture** [comment]
Also opens the door to defining per-field callbacks in the future to ensure we _always_ get what we want in the format we expect. That's future release though :-)

**Comment by DrewAPicture** [comment]
@pippinsplugins This is ready for review. It introduces about 25 new tests specifically for the _types_ of results returned for single and multiple field queries as well as non-fields queries.

**Comment by pippinsplugins** [comment]
This works great! I cannot break it.

💯

**Comment by DrewAPicture** [comment]
Merged #2264.
